Add a Ctrl-C (SIGINT) break-into-debugger handler.

Amiberry only declared setup_brkhandler() (xwin.h) and left the activation
block in real_main2() commented out. This defines setup_brkhandler(),
installing a SIGINT handler (sigbrkhandler) that calls activate_debugger(),
and enables the setup_brkhandler() call at startup so pressing Ctrl-C drops
the guest into the debugger. The start_debugger/debuggable() activation that
follows is enabled by the break-at-start patch.

Index: demoscene-toolchain/submodules/amiberry/src/main.cpp
===================================================================
--- demoscene-toolchain.orig/submodules/amiberry/src/main.cpp
+++ demoscene-toolchain/submodules/amiberry/src/main.cpp
@@ -11,6 +11,7 @@
 #include <cstdio>
 #include <cstdlib>
 #include <cstring>
+#include <csignal>
 
 #ifdef USE_OLDGCC
 #include <experimental/filesystem>
@@ -1572,6 +1573,33 @@ void check_error_sdl(const bool check, c
 	}
 }
 
+#ifdef DEBUGGER
+/* Ctrl-C (SIGINT) breaks into the debugger. */
+static void sigbrkhandler (int /*signum*/)
+{
+	activate_debugger ();
+#ifndef _WIN32
+	/* Reinstall in case of System V signal semantics. */
+	signal (SIGINT, sigbrkhandler);
+#endif
+}
+#endif
+
+void setup_brkhandler (void)
+{
+#ifdef DEBUGGER
+#ifndef _WIN32
+	struct sigaction sa{};
+	sa.sa_handler = sigbrkhandler;
+	sa.sa_flags = SA_RESTART;
+	sigemptyset (&sa.sa_mask);
+	sigaction (SIGINT, &sa, nullptr);
+#else
+	signal (SIGINT, sigbrkhandler);
+#endif
+#endif
+}
+
 static int real_main2 (int argc, TCHAR **argv)
 {
 	set_config_changed();
@@ -1695,12 +1723,11 @@ static int real_main2 (int argc, TCHAR *
 	init_m68k (); /* must come after reset_frame_rate_hack (); */
 
 	if (graphics_init (true)) {
-	// This never gets triggered anyway, debuggable always returns 0 in WinUAE
-//#ifdef DEBUGGER
-//		setup_brkhandler ();
+#ifdef DEBUGGER
+		setup_brkhandler ();
 //		if (currprefs.start_debugger && debuggable ())
 //			activate_debugger ();
-//#endif
+#endif
 		if (!init_audio ()) {
 			if (sound_available && currprefs.produce_sound > 1) {
 				write_log (_T("Sound driver unavailable: Sound output disabled\n"));
